File: src/contests/cg/puzzle/objectivec/1d-bush-fire.m
// 🎮 CodinGame Puzzle - 1d-bush-fire
// https://www.codingame.com/training/easy/1d-bush-fire

#import <Foundation/Foundation.h>
#include <string.h>

static char strip[1000006];

int main() {
    int n;
    scanf("%d", &n);
    for (int t = 0; t < n; t++) {
        scanf("%1000000s", strip);
        int len = strlen(strip);
        int drops = 0, j = 0;
        while (j < len) {
            if (strip[j] == 'f') {
                // Drop at j covers j, j+1, j+2 — skip past all 3.
                drops++;
                j += 3;
            } else {
                j++;
            }
        }
        printf("%d\n", drops);
    }
    return 0;
}

File: src/contests/cg/puzzle/vbnet/1d-bush-fire.vb
' 🎮 CodinGame Puzzle - 1d-bush-fire
' https://www.codingame.com/training/easy/1d-bush-fire

Imports System

Module Solution
    Sub Main()
        Dim n As Integer = Integer.Parse(Console.ReadLine().Trim())
        For t As Integer = 0 To n - 1
            Dim strip As String = Console.ReadLine().TrimEnd()
            Dim drops As Integer = 0, j As Integer = 0
            While j < strip.Length
                If strip(j) = "f"c Then
                    ' Drop at j covers j, j+1, j+2 — skip past all 3.
                    drops += 1
                    j += 3
                Else
                    j += 1
                End If
            End While
            Console.WriteLine(drops)
        Next
    End Sub
End Module
